docs/sphinx/examples/cpp/algorithms/bernstein_vazirani.cpp

```diff
@@ -0,0 +1,88 @@
+// Compile and run with:
+// ```
+// nvq++ bernstein_vazirani.cpp -o bv.x --target nvidia-mgpu
+// mpirun -np 4 ./bv.x
+// ```
+
+// This example is meant to demonstrate the cuQuantum
+// GPU-accelerated multi-GPU backend. A state vector simulation for
+// up to 32 qubits can be run on a single GPU. Larger state vectors
+// can be simulated using the multi-GPU backend.
+// The amount of resources required for the simulation doubles with
+// with each additional qubit.
+
+#include <bitset>
+#include <cudaq.h>
+#include <iostream>
+#include <random>
+
+template <int nrOfBits>
+std::bitset<nrOfBits> random_bits() {
+
+  std::bitset<nrOfBits> randomBits;
+
+  std::default_random_engine generator;
+  std::uniform_real_distribution<float> distribution(0.0, 1.0);
+
+  float randNum;
+  for (size_t i = 0; i < nrOfBits; i++) {
+    randNum = distribution(generator);
+    if (randNum < 0.5) {
+      randNum = 0;
+    } else {
+      randNum = 1;
+    }
+    randomBits.set(i, randNum);
```

Comment:
This is just a nit, but it caught my eye as a little strange to pass a float (`randNum`) into `randomBits.set()`. The following would shorten the code an avoid passing a float to that function.
```suggestion
  for (size_t i = 0; i < nrOfBits; i++) {
    if (distribution(generator) < 0.5)
      randomBits.set(i, 0);
    else
      randomBits.set(i, 1);
```
Of course, this would shorten it even more:
```suggestion
  for (size_t i = 0; i < nrOfBits; i++) {
    randomBits.set(i, distribution(generator) < 0.5 ? 0 : 1);
```